apps/bfd-pipeline/bfd-pipeline-shared-utils/src/main/java/gov/cms/bfd/pipeline/sharedutils/SamhsaUtil.java

```diff
@@ -562,12 +563,35 @@ public static Optional<SamhsaEntry> getSamhsaCode(Optional<String> code) {
         throw new RuntimeException("Cannot retrieve list of SAMHSA codes.");
       }
     }
-    if (samhsaMap.containsKey(code.get())) {
-      return Optional.of(samhsaMap.get(code.get()));
+    String normalizedCode = normalizeCode(code.get());
+
+    if (samhsaMap.containsKey(normalizedCode)) {
+      return Optional.of(samhsaMap.get(normalizedCode));
     }
     return Optional.empty();
   }
 
+  private static void normalizeSamhsaMapKeys() {
+    // Replace the original map with the normalized map
+    samhsaMap =
+        samhsaMap.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    entry -> normalizeCode(entry.getKey()),
+                    Map.Entry::getValue,
+                    (existing, replacement) -> existing));
+  }
+
+  private static String normalizeCode(String code) {
+    code = code.trim();
+    code = code.replaceFirst("\\.", "");
+    if (code.startsWith("MS-DRG ")) {
+      code = code.replace("MS-DRG ", "");
```

Comment:
Being (overly) paranoid again, but I think this is probably more correct if we're being strict about it.

```suggestion
      code = code.replaceFirst("MS-DRG ", "");
```